src/java/org/apache/cassandra/config/DefaultLoader.java

```diff
@@ -0,0 +1,123 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.config;
+
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.yaml.snakeyaml.error.YAMLException;
+import org.yaml.snakeyaml.introspector.FieldProperty;
+import org.yaml.snakeyaml.introspector.MethodProperty;
+import org.yaml.snakeyaml.introspector.Property;
+
+import static org.apache.cassandra.utils.FBUtilities.camelToSnake;
+
+public class DefaultLoader implements Loader
+{
+    @Override
+    public Map<String, Property> getProperties(Class<?> root)
+    {
+        Map<String, Property> properties = new HashMap<>();
+        for (Class<?> c = root; c != null; c = c.getSuperclass())
+        {
+            Field[] fields = c.getDeclaredFields();
+            if (fields != null)
+            {
+                for (Field f : fields)
+                {
+                    String name = camelToSnake(f.getName());
+                    int modifiers = f.getModifiers();
+                    if (!Modifier.isStatic(modifiers)
+                        && !f.isAnnotationPresent(JsonIgnore.class)
+                        && !Modifier.isTransient(modifiers)
+                        && Modifier.isPublic(modifiers)
+                        && !properties.containsKey(name))
+                        properties.put(name, new FieldProperty(f));
+                }
+            }
+        }
+        try
+        {
+            PropertyDescriptor[] descriptors = Introspector.getBeanInfo(root).getPropertyDescriptors();
+            if (descriptors != null)
+            {
+                for (PropertyDescriptor d : descriptors)
+                {
+                    String name = camelToSnake(d.getName());
+                    Method writeMethod = d.getWriteMethod();
+                    // if the property can't be written to, then ignore it
+                    if (writeMethod == null || writeMethod.isAnnotationPresent(JsonIgnore.class))
+                        continue;
+                    // if read method exists, override the field version in case get/set does validation
+                    if (properties.containsKey(name) && d.getReadMethod() == null)
+                        continue;
```

Comment:
Are there any tests that exercise these two conditional branches where we don't add a property?

Reply:
there are now!

Reply:
found one issue where getter is JsonIgnored